### ISSUE TITLE
Candidates::determineLocale regex matches wrongly

### ISSUE THREAD

**Issue by uwej711** [symfony-cmf/Routing#194]
Hi all,

I'm currently upgrading one of our projects and ran into an issue with 
https://github.com/symfony-cmf/routing/blob/master/src/Candidates/Candidates.php#L111

The regular expression there matches my url "/buchen" if I have "en" in the list of locales. I guess it should be 
```
if (preg_match('#^/('.implode('|', $this->locales).')(/|$)#', $url, $matches)) {
```
instead.

**Comment by uwej711** [comment]
I can do a PR against 1.4

**Comment by dbu** [comment]
ouch, thats quite bad! yes please do a pr and i will merge it to master if
its still present there

Am 08.05.2017 1:58 nachm. schrieb "Uwe Jäger" <notifications@github.com>:

**Comment by uwej711** [comment]
I would be happy if that would go in a 1.4.1 version as well ... spares me maintaining a fork.

**Comment by dbu** [comment]
fixed in #195
